reframe/core/config.py

```diff
@@ -217,3 +220,151 @@ def create_env(system, partition, name):
                 system.add_partition(part)
 
             self._systems[sys_name] = system
+
+
+def ppretty(value, htchar=' ', lfchar='\n', indent=4, current_depth=0):
+    nlch = lfchar + htchar * indent * (current_depth + 1)
+    if type(value) is dict:
+        if value == {}:
+            return '{}'
+        items = [
+            nlch + repr(key) + ': ' +
+            ppretty(value[key], htchar, lfchar, indent, current_depth + 1)
+            for key in value
+        ]
+        return '{%s}' % (','.join(items) + lfchar +
+                         htchar * indent * current_depth)
+    elif type(value) is list:
+        if value == []:
+            return '[]'
+        items = [
+            nlch + ppretty(item, htchar, lfchar, indent, current_depth + 1)
+            for item in value
+        ]
+        return '[%s]' % (','.join(items) + lfchar +
+                         htchar * indent * current_depth)
+    elif type(value) is tuple:
```

Comment:
```suggestion
    elif isinstance(value, tuple):
```

And put this check first, before the check for the `Sequence`.